packages/fpc.rb

```diff
@@ -3,56 +3,48 @@
+     armv7l: 'ec794516f488468c8375cfc16ae2ea5e7a9cd1dc41f67f9129c6f34992bfcae9',
+       i686: 'dd54485d5db294e4cd713debc7e09b3f00150b46574dfd7bf16b21e781428729',
+     x86_64: '1264a9601eb7258c0482cf53746c0b1ae3cb06871b6d7c4ffad135d62935abeb'
   })
 
-  def self.patch
-    system "sed -i 's,PREFIX=\"\$HOME/fpc-\$VERSION\",PREFIX=#{CREW_DEST_PREFIX},' install.sh"
-    system "sed -i 's,\$PREFIX/lib,#{CREW_DEST_LIB_PREFIX},' install.sh"
+  print_source_bashrc
+
+  def self.build
+    # https://gitlab.com/freepascal.org/fpc/source/-/issues/40841
+    # fpc isn't fooled by our docker emulation, and thus disables reading from the config because it thinks we're cross-compiling.
+    # See https://gitlab.com/freepascal.org/fpc/source/-/commit/898a4debcab843b2f6f5c29b795a517485582dfb
+    # Because we don't read from the config, we don't get the correct library paths.
+    # The correct thing to do here would be to find the fault in our docker setup that reveals we're running under emulation, and fix it.
+    # In the meantime, though, just pass the correct library path manually.
+    # Chromebrew messes things up if we don't pass this in one string, adding single quotes around OPT which break things.
+    system "make OVERRIDEVERSIONCHECK=1 OPT='#{"-Fl#{CREW_LIB_PREFIX}" unless ARCH.eql?('x86_64')} -k-L#{CREW_LIB_PREFIX}' all"
   end
 
   def self.install
-    system './install.sh'
-    FileUtils.ln_sf "#{CREW_LIB_PREFIX}/fpc/#{version}/ppcarm", "#{CREW_DEST_PREFIX}/bin/ppcarm"
-    FileUtils.mv "#{CREW_DEST_PREFIX}/man", CREW_DEST_MAN_PREFIX
+    system 'make', "PREFIX=#{CREW_DEST_PREFIX}", 'install'
+    compiler = { aarch64: 'ppcarm', armv7l: 'ppcarm', i686: 'ppc386', x86_64: 'ppcx64' }
+    FileUtils.ln_sf "#{CREW_PREFIX}/lib/fpc/#{version}/#{compiler[ARCH.to_sym]}", "#{CREW_DEST_PREFIX}/bin/#{compiler[ARCH.to_sym]}"
+    # Set up the default fpc config file.
+    system "#{CREW_DEST_PREFIX}/lib/fpc/#{version}/samplecfg #{CREW_DEST_PREFIX}/lib/fpc/#{version} #{CREW_DEST_PREFIX}/etc"
+    system "sed -i '/# searchpath for libraries/a\-Fl#{CREW_LIB_PREFIX}' #{CREW_DEST_PREFIX}/etc/fpc.cfg"
+    File.write 'fpcenv', <<~FPC_ENV_EOF
+      # Set PPC_CONFIG_PATH so that fpc finds fpc.cfg
+      export PPC_CONFIG_PATH=#{CREW_PREFIX}/etc
+    FPC_ENV_EOF
+    FileUtils.install 'fpcenv', "#{CREW_DEST_PREFIX}/etc/env.d/fpc", mode: 0o644
   end
 
   def self.postinstall
-    puts "\nType 'fp' to start.\n".lightblue
-  end
-
-  def self.remove
-    print 'Would you like to remove the config directories? [y/N] '
-    response = $stdin.gets.chomp.downcase
-    config_dirs = ["#{HOME}/.fpc.cfg", "#{HOME}/.config/fppkg.cfg", "#{HOME}/.fppkg"]
-    config_dirs.each do |config_dir|
-      case response
-      when 'y', 'yes'
-        FileUtils.rm_rf config_dir
-        puts "#{config_dir} removed.".lightred
-      else
-        puts "#{config_dir} saved.".lightgreen
-      end
-    end
+    ExitMessage.add "Type 'fp' to start.".lightblue
```

Comment:
We still want the user to have the option to remove the config directories if they still exist in this release.  If you're like me, I absolutely hate when I uninstall a package and it leaves residue behind.

Reply:
They are in `CREW_PREFIX/etc` now, so they'll be handled by crew remove.